integrating-elasticsearch-indexing-and-directus/index.md

```diff
@@ -0,0 +1,165 @@
+
+## Getting Elasticsearch Connection Details
+
+To connect to Elasticsearch the Cloud ID and API key credentials will be used. You can in Kibana of your Elastic Cloud deployment, click on **Setup guides** at the right of the header. On the navigated page, click on **Connect to the Elasticsearch API** card and you will see your Cloud ID.
+
+![A modal showing the Elasticsearch endpoint and Cloud ID which has been obscured](elastic-cloud-id.png)
+
+Now to get an API key, click on **Create and manage API keys,** then at the top-right of the navigated page click **Create API key.** Enter the name and click **Create API key** at the bottom-right of the sidebar.
+
+![A sidebar to create an API key in Elastic Cloud. It includes a name field with the value diectus-integration](create-elastic-api-key.png)
+
+Copy the displayed API key as well as the Cloud ID and paste them into the `api/.env` file in your Directus project.
+
+```shell
+ELASTIC_API_KEY=<your-api-key>
+ELASTIC_CLOUD_ID=<your-cloud-id>
+```
+
+## Creating a Collection in Directus
+
+The collection to be created is the one that needs to be in sync with an Elasticsearch index. In your Directus Admin App, navigate to **Settings -> Data Model** and create a new collection called `books` with a text input called `title` and a textarea field called `description`.
+
+![The data model of the books collection in Directus Admin app.](books-collection.png)
+
+## Creating a Custom API Hooks Extension
+
+As stated in the [docs](https://docs.directus.io/extensions/hooks.html) Custom API Hooks allow running custom logic when a specified event occurs within your project. These events include item creation, deletion, etc.
+
+There are several types of hooks used to define the custom logic, such as [action](https://docs.directus.io/extensions/hooks.html#action), [filter](https://docs.directus.io/extensions/hooks.html#init), [init](https://docs.directus.io/extensions/hooks.html#schedule), etc. The specific hooks to use depend on the operation to be carried out. For this use case where an Elasticsearch index is to be updated when an item is created, updated, and deleted in a Directus collection to keep the index and collection in sync, the `action` hook will do the trick.
+
+To create the hook extension boilerplate, in the terminal, enter the following command in the root directory of your Directus project with the following options:
+
+```shell
+npx create-directus-extension@latest
+? Choose the extension type: hook
+? Choose a name for the extension: directus-extension-elasticsearch
+? Choose the language to use: javascript
+? Auto install dependencies? (Y/n) y
+```
+
+After that, navigate into the created hook directory and install the Elasticsearch Javascript Client:
+
+```shell
+cd directus-extension-elasticsearch
+npm install @elastic/elasticsearch
+```
```

Comment:
````suggestion
## Creating a Custom Hook Extension

Custom hook extensions allow running custom logic triggered by an event emitted by Directus. These events includes key database events like item creation and deletion.

For this use case where an Elasticsearch index is to be updated when an item is created, updated, or deleted in a Directus collection to keep the index and collection in sync, you will use an `action` hook, which will run after an event has occurred.

In the terminal, enter the following command in the your mounted extensions directory:

```shell
npx create-directus-extension@latest
? Choose the extension type: hook
? Choose a name for the extension: elasticsearch-sync
? Choose the language to use: JavaScript
? Auto install dependencies? (Y/n) y
```

After that, navigate into the created hook directory and install the Elasticsearch JavaScript Client:

```shell
cd elasticsearch-sync
npm install @elastic/elasticsearch
```
````

Reply:
Changing the directory name as we dropped the need to prefix extensions with directus-extension in v10.10.0 and I don't want to confuse matters by including it here.

Reply:
@phazonoverload From your suggestions I noticed that you have replaced the process of creating the extension boilerplate in the root directory and then moving it to the `api/extensions` directory with a pre-mounted extensions directory as stated in line 50.
Shouldn't there be a link or description on how to mount an extensions directory?

Reply:
Okay